spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -587,54 +521,36 @@ private MessageRecoverer determineRecoverer(String name, RabbitPropertiesAccesso
 		}
 	}
 
-	@Override
-	public void bindProducer(final String name, MessageChannel moduleOutputChannel,
-			Properties properties) {
-		Assert.isInstanceOf(SubscribableChannel.class, moduleOutputChannel);
-		RabbitPropertiesAccessor accessor = new RabbitPropertiesAccessor(properties);
-		if (name.startsWith(P2P_NAMED_CHANNEL_TYPE_PREFIX)) {
-			validateProducerProperties(name, properties, SUPPORTED_NAMED_PRODUCER_PROPERTIES);
-		}
-		else {
-			validateProducerProperties(name, properties, SUPPORTED_PRODUCER_PROPERTIES);
-		}
-		if (!bindNewProducerDirectlyIfPossible(name, (SubscribableChannel) moduleOutputChannel, accessor)) {
-			if (this.logger.isInfoEnabled()) {
-				this.logger.info("declaring queue for outbound: " + name);
-			}
-			AmqpOutboundEndpoint queue = this.buildOutboundEndpoint(name, accessor, determineRabbitTemplate(accessor));
-			doRegisterProducer(name, moduleOutputChannel, queue, accessor);
-		}
-	}
-
 	private AmqpOutboundEndpoint buildOutboundEndpoint(final String name, RabbitPropertiesAccessor properties,
 			RabbitTemplate rabbitTemplate) {
 		String prefix = properties.getPrefix(this.defaultPrefix);
-		String queueName = applyPrefix(prefix, name);
+		String exchangeName = applyPrefix(prefix, name);
 		String partitionKeyExtractorClass = properties.getPartitionKeyExtractorClass();
 		Expression partitionKeyExpression = properties.getPartitionKeyExpression();
-		TopicExchange exchange = new TopicExchange(queueName);
-		declareExchange(queueName, exchange);
+		TopicExchange exchange = new TopicExchange(exchangeName);
+		declareExchange(exchangeName, exchange);
 		AmqpOutboundEndpoint endpoint = new AmqpOutboundEndpoint(rabbitTemplate);
 		endpoint.setExchangeName(exchange.getName());
+		String baseQueueName = exchangeName + ".default";
 		if (partitionKeyExpression == null && !StringUtils.hasText(partitionKeyExtractorClass)) {
-			Queue queue = new Queue(queueName, true, false, false, queueArgs(properties, queueName));
-			declareQueue(name, queue);
-			autoBindDLQ(name, properties);
-			endpoint.setRoutingKey(queueName);
-			org.springframework.amqp.core.Binding binding = BindingBuilder.bind(queue).to(exchange).with(queueName);
-			declareBinding(queueName, binding);
+			Queue queue = new Queue(baseQueueName, true, false, false, queueArgs(properties, baseQueueName));
+			declareQueue(baseQueueName, queue);
+			autoBindDLQ(baseQueueName, baseQueueName, properties);
+			endpoint.setRoutingKey(name);
+			org.springframework.amqp.core.Binding binding = BindingBuilder.bind(queue).to(exchange).with(name);
```

Comment:
There is a slight difference between the way the default queue is bound (with the routing key matching the `name`) and the way explicitly named consumer groups are bound (hash). While this may not have any practical consequences (the outbound producers will always use the proper routing key), I think it would be better if we were consistent, and avoid some subtle bugs down the road.